src/util/db.rs

```diff
@@ -264,13 +264,58 @@ macro_rules! impl_byte_array_from_column {
     };
 }
 
+/// Load the path of the database from the connection
+fn get_db_path(conn: &Connection) -> Result<String, Error> {
+    let sql = "PRAGMA database_list";
+    let path: Result<Option<String>, sqlite_error> =
+        conn.query_row_and_then(sql, NO_PARAMS, |row| row.get(2));
+    match path {
+        Ok(Some(path)) => Ok(path),
+        Ok(None) => Ok("<unknown>".to_string()),
+        Err(e) => Err(Error::SqliteError(e)),
+    }
+}
+
+/// Generate debug output to be fed into an external script to examine query plans.
+/// TODO: it uses mocked arguments, which it assumes are strings. This does not always result in a
+/// valid query.
+#[cfg(test)]
+fn log_sql_eqp(conn: &Connection, sql_query: &str) {
+    if std::env::var("BLOCKSTACK_DB_TRACE") != Ok("1".to_string()) {
+        return;
+    }
+
+    let mut parts = sql_query.clone().split(" ");
+    let mut full_sql = if let Some(part) = parts.next() {
+        part.to_string()
+    } else {
+        sql_query.to_string()
+    };
+
+    while let Some(part) = parts.next() {
```

Comment:
I'm not sure what you're asking?  The code is constructing a new string because trying to replace `?X` with `"mock_arg"` would in effect be doing the same thing.